### DISCUSSION TITLE
Feature Request: Tick Trigger

### DISCUSSION THREAD

**Discussion by cjohnsto-nz** [jcorporation/myMPD#595] [Ideas]
I'd like to see a tick trigger, that basically runs every few seconds while myMPD is running. I've got myMPD set up to control my amplifier volume through scripting. If the remote is used to change volume, they go out of sync. The best I have at the moment is to use the mpd_player trigger to wait until a song ends, or someone interacts with the player to sync up.

I would also like this, because it means I could quickly and automatically pause/stop myMPD if the amp is turned off.

**Comment by jcorporation** [comment]
This could be implemented. I think such a trigger should be default disabled to not waste resources.

For now you can emulate this:
1. add a trigger for mpd_connected and attach a script that joins a mpd channel
2. add a trigger for a channel message that executes your wanted actions
3. use mpc in a cronjob to send channel messages

**Reply by cjohnsto-nz** [reply]
That would be fantastic. If the ticker could be stopped and started through the API, and have the interval changed, it would be even better. That way you could run it automatically when myMPD starts playing, and stop the music and the ticker when the output or power of the amp changed.

**Comment by jcorporation** [comment]
Alternatively you can use the mympd-script command to start a script from a cronjob.

**Reply by cjohnsto-nz** [reply]
I did think about that. It is doable, but cron has a minimum resolution of a minute.

**Comment by jcorporation** [comment]
Thinking again about your request: the myMPD timer function should handle your scenario perfectly.

**Reply by cjohnsto-nz** [reply]
That's true, but I couldn't figure out how to start and stop a repeating timer without specifying a start date and time.

**Reply by jcorporation** [reply]
Set start to now and it should work immediately. The interval could be not defined as fine granular you want, but this is only a small ui/api change.

I think we should go this way, it’s much easier to implement and uses already implemented functions.
